staging/src/k8s.io/client-go/tools/cache/controller_test.go

```diff
@@ -298,13 +298,11 @@ func TestHammerController(t *testing.T) {
 	time.Sleep(100 * time.Millisecond)
 	cancel()
 
-	// Before we permanently lock this mutex, we have to be sure
-	// that the controller has stopped running. At this point,
-	// all goroutines should have stopped. Leak checking is
-	// done by TestMain.
+	// Wait until the controller has fully stopped running.
+	// At this point, all goroutines should have stopped, and
+	// leak checking is done by TestMain.
 	controllerWG.Wait()
 
-	outputSetLock.Lock()
 	t.Logf("got: %#v", outputSet)
```

Comment:
if you want to keep the log you need to maintain the lock because it is accessing the `outputSet` variable, however, now I don't understand the comment, why do we want to perma lock the handler, if that happens then the goleak will fail as in the reported issue

@liggitt  do you have context on this test? I'm continuously doing wrong assumptions here

Reply:
> @liggitt do you have context on this test? I'm continuously doing wrong assumptions here

no, sorry

Reply:
I've removed the log. Let me know if you think the perma lock is actually needed. Again, sorry for all the back-and-forth.

Reply:
Ok, I think I understand it now, the Lock is correct based on @pohly's commit https://github.com/kubernetes/kubernetes/commit/8e1403563a60f3b7a258e3bbb64b5c3a7f6548fb

Once the controller is finished there must not be  more processing, so we lock on purpose because that will detect a goroutine leak ... so maybe the problem is not the test and is the logic in the controller that causes the flake?